File: backend/src/NeoantigenPipeline.Api/Models/Patient.cs
namespace NeoantigenPipeline.Api.Models;

public class Patient
{
    public string Id { get; set; } = "";
    public string Name { get; set; } = "";
    public string? Notes { get; set; }
    public string? CancerType { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string? ReferenceGenome { get; set; }
}

File: frontend/src/types/patient.ts
export interface Patient {
  id: string;
  name: string;
  notes?: string;
  cancerType?: string;
  createdAt: string;
  updatedAt: string;
  referenceGenome?: string;
}

export interface PatientSummary {
  id: string;
  name: string;
  cancerType?: string;
  createdAt: string;
  completedSteps: number;
  totalSteps: number;
  furthestStepId?: string;
  totalDiskBytes: number;
}

export interface CreatePatientRequest {
  name: string;
  notes?: string;
  cancerType?: string;
  referenceGenome?: string;
}

export interface UpdatePatientRequest {
  name?: string;
  notes?: string;
  cancerType?: string;
}
